File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dir = {}
    answer = 1

    for a, b in clothes:
        if b not in dir:
            dir[b] = [a]
        else:
            dir[b].append(a)

    for key in dir:
      answer *= (len(dir[key]) + 1)
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;

        HashMap<String, Integer> map = new HashMap<>();
        for (int i = 0; i < clothes.length; i++) {
            String type = clothes[i][1];
            if (map.containsKey(type)) {
                map.put(type, map.getOrDefault(type, map.get(type)) + 1);
            } else {
                map.put(type, map.getOrDefault(type, 0) + 1);
            }
        }

        for (int cnt : map.values()) {
            answer *= (cnt + 1);
        }

        return answer - 1;
    }
}